### ISSUE TITLE
Crash when using Spark write to CK

### ISSUE THREAD

**Issue by pan3793** [housepower/ClickHouse-Native-JDBC#181]
### JDBC version in pom.xml
2.3-stable shaded version

### ClickHouse version
20.8-lts

### Error  logs
```
Lost task 153.2 in stage 9.0 (TID 265, cdh-worker003, executor 34): com.github.housepower.jdbc.ClickHouseSQLException: DB::ExceptionDB::Exception: Unexpected packet Data received from client. Stack trace:

0. Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x18ba03d0 in /usr/bin/clickhouse
1. DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0xe606dad in /usr/bin/clickhouse
2. DB::TCPHandler::receiveUnexpectedData() @ 0x1626ab89 in /usr/bin/clickhouse
3. ? @ 0x1626d281 in /usr/bin/clickhouse
4. DB::TCPHandler::runImpl() @ 0x1626e83f in /usr/bin/clickhouse
5. DB::TCPHandler::run() @ 0x1626f6e0 in /usr/bin/clickhouse
6. Poco::Net::TCPServerConnection::start() @ 0x18abe23b in /usr/bin/clickhouse
7. Poco::Net::TCPServerDispatcher::run() @ 0x18abe6cb in /usr/bin/clickhouse
8. Poco::PooledThread::run() @ 0x18c3d1a6 in /usr/bin/clickhouse
9. Poco::ThreadImpl::runnableEntry(void*) @ 0x18c385a0 in /usr/bin/clickhouse
10. start_thread @ 0x7dd5 in /usr/lib64/libpthread-2.17.so
11. __clone @ 0xfe02d in /usr/lib64/libc-2.17.so

	at com.github.housepower.jdbc.protocol.ExceptionResponse.readExceptionFrom(ExceptionResponse.java:32)
	at com.github.housepower.jdbc.protocol.RequestOrResponse.readFrom(RequestOrResponse.java:38)
	at com.github.housepower.jdbc.connect.PhysicalConnection.receiveResponse(PhysicalConnection.java:95)
	at com.github.housepower.jdbc.connect.PhysicalConnection.receiveEndOfStream(PhysicalConnection.java:87)
	at com.github.housepower.jdbc.ClickHouseConnection.sendInsertRequest(ClickHouseConnection.java:159)
	at com.github.housepower.jdbc.statement.ClickHousePreparedInsertStatement.close(ClickHousePreparedInsertStatement.java:94)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:675)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$$anonfun$saveTable$1.apply(JdbcUtils.scala:834)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$$anonfun$saveTable$1.apply(JdbcUtils.scala:834)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$28.apply(RDD.scala:935)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$28.apply(RDD.scala:935)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2121)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2121)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$11.apply(Executor.scala:407)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1408)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:413)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by pan3793** [comment]
It can reproduce using 2.4.1 write to clickhouse 20.3

**Comment by yangmeng66** [comment]
@pan3793 Hello, I also used spark to import clikchouse and encountered the same problem. The version of clickhouse I used is: 21.8.3.44 native-jdbc version is 2-5-6, how did you solve it in the end? Just change Is the version of jdbc? 2.3 ->> 2.4.1 ??

![image](https://user-images.githubusercontent.com/51857567/132969430-48556d51-febc-4b3f-b654-16909bb22c1f.png)

**Comment by pan3793** [comment]
You can also use `clickhouse-integration-spark` with the official clickhouse JDBC driver or `clickhouse4j`.

**Comment by yangmeng66** [comment]
hi, first of all thank you for your reply, it's like this,
1. I used the official JDBC driver at first, but "Broken pipe" or "connect closed" often appeared during use, and finally I gave up using it.
2. Next, I used native-jbdc to import and found that this is very stable and very fast.
3. But recently I kept reporting this error when importing a table with 44 attributes. I also tried the official jdbc driver and clickhouse4j. Although this error is not reported, it is very slow.
4. The clickhouse-integration-spark you mentioned supports , but our company now uses version 2.1.0 of Spark

To summarize: I use native-jdbc to import 14-attribute tables and there is no problem, but this error occurs when I import this 44-attribute table

**Comment by pan3793** [comment]
Could you please provide the reproducible code and data?

**Comment by yangmeng66** [comment]
Yes, can I reply to you tomorrow?, Today is the weekend, I wish you a happy weekend

**Comment by pan3793** [comment]
Yeah

**Comment by yangmeng66** [comment]
Hi, pan, I tried to reproduce the error locally using 5000 records in the database, but the error cannot be reproduced locally. Only when a large amount of data is imported to the server, the error is caused by Spark logs.

![image](https://user-images.githubusercontent.com/51857567/133352252-895d4d17-2aaa-4449-8799-e9c68865c06e.png)
![image](https://user-images.githubusercontent.com/51857567/133352227-2d40dfd1-a92e-4a04-a1a4-235a5834af4c.png)


备注: 
1 我使用的驱动版本
![image](https://user-images.githubusercontent.com/51857567/133352313-71fb0c57-2d8e-4197-a245-4f9b97e27856.png)
2. clickhouse版本是: 21.8.3.44

**Comment by luozhiyuan55** [comment]
hello, how do you handle the problem finally?
